### ISSUE TITLE
Step doesn't display after redirect using hash

### ISSUE THREAD

**Issue by jgolieb** [sorich87/bootstrap-tour#493]
The first step of my tour is at path "/#/".  If I am at a different location/route in my site (same path,  different hash) and start the tour, the redirect to /#/ works, but the step does not display.  This is because after the redirect, _isJustPathHashDifferent() returns false causing showStepHelper() to just return.  I commented out the check of _isJustPathHashDifferent() and everything seems to work.  I'm not sure what scenario this check is there for though.

**Comment by esoyke** [comment]
@jgolieb Did you ever find a workaround for this?

**Comment by jgolieb** [comment]
As I stated, my workaround was to comment out the lines that check the path using _isJustPathHashDifferent().  But I assume that was originally there for some reason, so I'm not really happy with this hack.

**Comment by esoyke** [comment]
@jgolieb Ha, sorry I misread, you DID find a workaround. Yeah, hacks are never welcome even when they appear to solve your problem. I was having weird side-effects with my multi-page tour and ended up switching over to angular-bootstrap-tour instead. It seems to be working ok but loses some of what I liked in the standard bootstrap-tour plugin (a centralized configuration of steps, ability to specify a nested element as the target), and is proving to introduce some challenges of its own.

**Comment by esoyke** [comment]
I found that even with this hack*, I still had to cheat a bit by adding an intermediate 'hidden' dummy step when switching pages in my Angular SPA (note my hash path includes a !, but I think we essentially had the same issue). I set it to orphan=true as to not be skipped if not already visible. The duration wasn't really important, but setting a delay on the next (real) step was. I went a little further and set the dummy step to be invisible to avoid the visual flicker. The delay was trial and error, I imagine on a slow browser you run the risk of the element still not being visible, figured 200ms was unnoticeable yet enough time for it to get rendered:

```
{
    path: "/#!/",
    element: "#tour-step1Element",
    title: "Step 1",
    content: "Step one's content..."
},
// a dummy step when you are switching to another page and the element isn't yet visible
{
    path: "/#!/page2",
    element: "#tour-dummyElement",
    duration: 1,
    orphan: true,
    template: "<div style='display: none'></div>"
},
{
    path: "/#!/page2",
    element: "#tour-step2Element",
    delay: 200,
    title: "Step 2",
    content: "Step 2 content..."
},
```

\* I had originally been using an older release of bootstrap-tour, as I had originally tried the angular-bootstrap-tour library which has a dependency on an older BT release, and didn't see the code you were referring to. When I updated my bower to pull v0.10.2, it actually pulled v0.10.1, even though Bower said it pulled v0.10.2. Weird. Had to manually copy 0.10.2 from the site then commented out lines 312-314 as you did. Thanks again.

**Comment by jgolieb** [comment]
This is going off on a tangent from the original issue, but I actually addressed the problem of waiting for a not-yet-loaded element in a more direct way.  I added an additional waitFor property to my steps and set it to the selector of the element that I actually want the popover attached to but that may not be loaded yet.  I set the element property to the selector of an element that will always be immediately loaded (such as "body").  Then, in the default onShown handler, I hide the popover created for element until waitFor gets added to the DOM.  I check for this using setInterval to check for the waitFor element's existence every 100 milliseconds.  Once it is there, I move the popover to the waitFor element and show it.  It would be great if this functionality were built in.
